### DISCUSSION TITLE
Domoticz Daikin backwards compatibility integration

### DISCUSSION THREAD

**Discussion by Lieuwe** [revk/xESP32-Faikout#236] [Q&A]
I'm trying to get the Faikin module to work in Domoticz, but the standard Daikin plugin (which allows the control of the old local Daikin API) is not working.
And the MQTT Autodiscovery is not working correctly. It is only adding temperature sensors, frequencies and a few not working switches. This may be a fault of the domoticz MQTT autodiscovery.
I tried MQTT Mapper plugin, but i'm not good at coding or understanding MQTT messages.

Is it possible to get the original Daikin plugin to work? I would like to have a few command controls in Domoticz: Power, Fanspeed and Setpoint

Plugin location of the original Daikin control in domoticz:
https://github.com/domoticz/domoticz/blob/development/hardware/Daikin.cpp

**Comment by revk** [comment]
I did try and emulate the old web interface used by other tools like this, and another contributor added a lot more, which I understood were enough to get this working. But I am not sure of how this tool works exactly, someone else may know.

If we can find any specific details of what is missing on the legacy URLs, I am happy to fix.

**Comment by Lieuwe** [comment]
Well, looking at the code there is maybe a small change to be made to the Faikin legacy http api.
I have a second Daikin up and running and i'm controlling that aircon with the legacy daikin plugin/api.
Reply from original Daikin /aircon/get_control_info:
ret=OK,pow=0,mode=4,adv=,stemp=19.5,shum=0,dt1=25.0,dt2=M,dt3=25.0,dt4=19.5,dt5=19.5,dt7=25.0,dh1=AUTO,dh2=50,dh3=0,dh4=0,dh5=0,dh7=AUTO,dhh=50,b_mode=4,b_stemp=19.5,b_shum=0,alert=255,f_rate=5,f_dir=3,b_f_rate=5,b_f_dir=3,dfr1=5,dfr2=5,dfr3=5,dfr4=5,dfr5=5,dfr6=5,dfr7=5,dfrh=5,dfd1=0,dfd2=0,dfd3=0,dfd4=3,dfd5=3,dfd6=0,dfd7=0,dfdh=0,dmnd_run=0,en_demand=1

Reply from Faikin:
ret=ok,pow=1,mode=4,adv=,stemp=17.5,shum=0,dt1=17.5,dt2=17.5,dt3=17.5,dt4=17.5,dt5=17.5,dt7=17.5,dh1=0,dh2=0,dh3=0,dh4=0,dh5=0,dh7=0,dhh=0,b_stemp=17.5,b_shum=0,alert=255,f_rate=6,f_dir=0,dfr1=0,dfr2=0,dfr3=0,dfr4=0,dfr5=0,dfr7=0,dfrh=0,dfd1=0,dfd2=0,dfd3=0,dfd4=0,dfd5=0,dfd7=0,dfdh=0,dmnd_run=0,en_demand=0

Small snippet of Daikin plugin code:
	if (sResult.find("ret=OK") == std::string::npos)
	{
		Log(LOG_ERROR, "Error getting data (check IP/Port)");
		return;
	}

Looking at it, the ret=ok needs to be ret=OK.
I don't know if capital letters make a difference, but knowing a little bit of linux or coding it are different characters, so it may be a big difference.

**Reply by revk** [reply]
Ooh, that is easy to try, hang on. I'll do a new beta.

Try now.

**Reply by Lieuwe** [reply]
And the devices are created. It only outputs 2 errors:
2024-02-09 16:15:46.651 Error: Perfera: Invalid data received
2024-02-09 16:15:46.761 Error: Perfera: Invalid data received
What this invalid data is, i don't know. Maybe i will investigate later.

But for now the switches are working!
![image](https://github.com/revk/ESP32-Faikin/assets/2933745/93b3cff8-9f13-45f1-8844-7cb7ecc63e77)

Thank you very much for this speedy solution!!

**Comment by revk** [comment]
If you find out what the issue is, let me know, should be easy to change.

**Reply by Lieuwe** [reply]
Yes, will definitely do!
It is a check of number of values in a returned string. Will check and compare later and report back which strings should have more values.

**Reply by Lieuwe** [reply]
Found the culprits:

/common/basic_info
Minimum number values: 8
Daikin return:
ret=OK,type=aircon,reg=eu,dst=1,ver=1_14_88,rev=8A618141,pow=0,err=0,location=0,name=%53%74%79%6c%69%73%68,icon=1,method=polling,port=30050,id=xxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx,pw=,lpw_flag=0,adp_kind=3,pv=3.30,cpv=3,cpv_minor=20,led=1,en_setzone=1,mac=xxxxxxxxxxxx,adp_mode=run,en_hol=0,ssid1=xxxxxxxxxx,radio1=-44,ssid=DaikinAPxxxxxx,grp_name=,en_grp=0
Faikin return:
ret=ok,type=aircon,reg=eu,mac=Fxxxxxxxxxxx,ssid=xxxxxxxxxxx
-> add 3 values

/aircon/get_year_power_ex
Minimum number values: 4
Daikin reply
ret=OK,curr_year_heat=3468/127/0/0/0/0/0/0/0/0/0/0,prev_year_heat=2673/2245/2352/970/391/36/0/0/0/570/1851/3132,curr_year_cool=0/0/0/0/0/0/0/0/0/0/0/0,prev_year_cool=0/0/0/0/0/546/494/320/161/4/0/0
Faikin reply:
ret=OK
->add 3 values

**Comment by revk** [comment]
OK faikin will not return `ret=ok` anywhere now, not on latest beta, I changed to `OK` for you.

I could fill in some stuff for `basic_info`, is `pow` the power on?

The `get_year_power_ex` I doubt I can do a lot other than maybe all zeros.

**Reply by Lieuwe** [reply]
Thank you!
pow is indeed power on (0 or 1)
the get_year_power_ex is fine with all zero's.
It's just to get rid of the errors in the plugin. I never saw any results from the year_power_ex.

**Reply by revk** [reply]
OK try latest beta...

**Reply by Lieuwe** [reply]
Working, no more errors!'
Thank you very much for this swift updates!
You can now add to the wiki that it is compatible with the Domoticz Daikin plugin!
